src/shogun/util/factory.h

```diff
@@ -65,7 +65,7 @@ namespace shogun
 		return features;
 	}
 
-	CFeatures* features(CFile* file, EPrimitiveType primitive_type = PT_FLOAT64)
+	CFeatures* features(CFile* file, machine_int_t primitive_type = PT_FLOAT64)
```

Comment:
this is more flexibile and we also need it to pass primitives from swig

Reply:
I am not sure I understand why you want that change? This effectively bypasses any type checking that swig code could do at runtime (passed a valid enum type)

Reply:
I understand now